tests/test_run_config_generator.py

```diff
@@ -0,0 +1,184 @@
+        Test Default Single Model:  
+            - No CLI options specified
+        
+       
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs
+
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def test_default_config_two_models(self):
+        """
+        Test Default Two Models:
+            - No CLI options specified
+
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model0
+                - my-model1
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**2
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def test_default_config_four_models(self):
+        """
+        Test Default Four Models:
+            - No CLI options specified
+
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model0
+                - my-model1
+                - my-model2
+                - my-model3
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**4
+
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def _run_and_test_run_config_generator(self,
+                                           yaml_content,
+                                           expected_config_count,
+                                           pa_cli_args=None):
```

Comment:
this third parameter isn't used in any calls to this function--why is it included?

Reply:
I originally copied this code over from my testing of PerfAnalyzerConfig. I'll remove this parameter.